app/queries/content_dependents.rb

```diff
@@ -0,0 +1,50 @@
+module Queries
+  class ContentDependents
+    def initialize(content_id:, fields: [], link_sets: nil, dependent_lookup: Queries::GetDependents.new)
+      @content_id = content_id
+      @fields = fields
+      @dependent_lookup = dependent_lookup
+      @link_sets = link_sets
+    end
+
+    def call
+      return [] unless has_dependants?
+      dependent_lookup.call(content_id, link_types)
+    end
+
+  private
+
+    attr_reader :content_id, :fields, :dependent_lookup
+
+    def has_dependants?
+      fields.any? do |field|
+        fields_that_require_dependent_lookup.include?(field)
+      end
+    end
+
+    def fields_that_require_dependent_lookup
+      link_sets.values.flatten.uniq
+    end
+
+    def link_types
+      link_sets.select(&method(:changed_fields_include?)).keys
+    end
+
+    def changed_fields_include?(_, link_set_field)
+      fields.any? { |field| link_set_field.include?(field) }
+    end
```

Comment:
`link_set_field` here is actually an array of fields, so it might help to name it as a plural.

I haven't seen many people using the method-proc syntax here (`&method`). I quite like it, but it might be unfamiliar to a lot of people.

I think you might be able to simplify this a bit by using the intersect operator `&`, e.g.

```ruby
def link_types
  link_sets.select { |_, expansion_fields| (fields & expansion_fields).any? }.keys
end
```

(This is untested.)